assignments/A2/Nim_Game.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Scanner;
+
+public class Nim_Game{
+	
+	
+	public static void main(String[] args){
+		nim();
+	}
+	
+	public static void nim(){
+		
+		String player1, player2;
+		int loopCondition =0;
+		
+		// creates random numbers for the piles
+		int A = (int)(Math.random()*5 + 1),
+				B = (int)(Math.random()*5 + 1),
+				C = (int)(Math.random()*5 + 1);
+		
+		int[] thePile = {A, B, C};
+		
+		Scanner input = new Scanner(System.in);
+		
+		// take in names from users
+		System.out.print("Player 1, enter your name: ");
+		player1 = input.nextLine();
+		
+		System.out.print("Player 2, enter your name: ");
+		player2 = input.nextLine();
+		
+		
+		
+		//Display the piles
+		System.out.printf("\nA: %d\tB: %d\tC: %d\n", thePile[0], thePile[1], thePile[2]);
+		
+		do{
+			thePile = nimgame(player1, player2, thePile);
+			loopCondition = thePile[0]+thePile[1]+thePile[2];
+			if(loopCondition == 0)
+				break;
+			thePile = nimgame(player2, player1, thePile);
+
+			loopCondition = thePile[0]+thePile[1]+thePile[2];
+		}
+		while(loopCondition != 0);
+		
+	}	
+	
+	public static int[] nimgame(String currentPlayer, String nextPlayer, int[] thePile){
```

Comment:
This and other methods below will only be called within this class, so they can be `private`